In [3]:
import numpy as np

**numpy has been imported into jupyter to carry out some numeric computations.**

In [9]:
rng = np.random.default_rng()

**rng has been defined as a Numpy Random Number Generator, this creates a default random generator instance
which can be used to generate random numbers.**

In [11]:
scores = rng.integers(100, size=50)
scores

array([25, 95, 79, 32, 63, 79,  6, 26, 62, 36,  4, 84, 26, 78, 68, 46, 34,
       60, 21, 87, 50, 88, 53, 13, 27, 98, 42, 80, 45, 21, 92, 56, 84, 57,
       85, 79, 75, 99, 53, 72, 16, 97, 73, 14, 45, 54, 58, 19, 84, 38])

***for the variable scores, a random number between (0-100) has been generated, with a specific range of (50).***

In [14]:
highest_num = np.max(scores)
highest_num

np.int64(99)

***from the numbers generated, the highest number is (99)***

In [15]:
lowest_num = np.min(scores)
lowest_num

np.int64(4)

***from the generated numbers, the lowest number is (4)***

In [16]:
scores_average = np.mean(scores)
scores_average

np.float64(55.56)

**within the (50) numbers that was randomly generated, the average is (55.56)**

In [17]:
scores_above25 = scores[scores > 25]
scores_above25

array([95, 79, 32, 63, 79, 26, 62, 36, 84, 26, 78, 68, 46, 34, 60, 87, 50,
       88, 53, 27, 98, 42, 80, 45, 92, 56, 84, 57, 85, 79, 75, 99, 53, 72,
       97, 73, 45, 54, 58, 84, 38])

***the above numbers are numbers within the generated scores that are greater(>) than (25)***